### PR TITLE
Support some options via data attributes

### PR DESCRIPTION
**Pull request by huoxito**
now `content`, `openOn` and `position` can be set via data attributes

as per https://github.com/HubSpot/tooltip/pull/45

### CODE REVIEW

File: src/js/drop.js [hunk 1]
```diff
@@ -124,6 +124,23 @@ function createContext(options={}) {
         throw new Error('Drop Error: You must provide a target.');
       }
 
+      var dataPrefix = 'data-' + drop.classPrefix;
```
Comment (geekjuice) [inline src/js/drop.js]: Mind changing this and the variables below to use ES6 string interpolation? e.g.
```javascript
const dataPrefix = `data-${drop.classPrefix}`;
```

File: src/js/drop.js [hunk 2]
```diff
@@ -124,6 +124,23 @@ function createContext(options={}) {
...
+
+      var contentAttr = this.target.getAttribute(dataPrefix);
```
Comment (geekjuice) [inline src/js/drop.js]: Seems like we can DRY this up a little and put these in a loop. That way, we can add support for additional options easily.

**Comment by huoxito** [comment]
thanks for the feedback @geekjuice I've updated the PR pls take a look when you have a chance

**Comment by geekjuice** [comment]
Sorry for the delay :crying_cat_face: 

I'm back from hiatus so I should be more responsive going forward.

Everything looks great, but I did have one question: If a data-attribute is available, should it override anything set programmatically vis constructor? The changes here seem to hint that anything set on the DOM element directly data-attribute has higher precedence than options set from using the constructor API.

Just wanted to confirm if this was intended or if we should still give options set by the constructor API precedence. I actually think the current implementation makes more sense, but just wanted to confrim before merging. :+1:

**Comment by huoxito** [comment]
hey @geekjuice no worries thanks for getting back o/

and yeah the idea is any data attribute would override constructor level settings it seems to make sense (same approach with inline css styles for example). Let me know if you'd like this other way

cheers

**Comment by geekjuice** [comment]
Awesome. Just wanted to confirm that. LGTM :tada: :+1:

**Pull request merged**